libs/ogl/shader_program.cc

```diff
@@ -101,18 +101,27 @@ ShaderProgram::compile_shader (GLuint shader_id, std::string const& code)
 bool
 ShaderProgram::try_load_all (std::string const& basename)
 {
-    std::string vert_filename = basename + ".vert";
-    std::string geom_filename = basename + ".geom";
-    std::string frag_filename = basename + ".frag";
+    return try_load_specific(basename, basename, basename);
+}
+
+bool
+ShaderProgram::try_load_specific (std::string const& vert_basename,
+    std::string const& geom_basename, std::string const& frag_basename)
+{
+    std::string vert_filename = vert_basename + ".vert";
```

Comment:
For this method, we should not append the extension. I.e., construct the filename from the basename in `try_load_all`, and rename the arguments to `vert_filename`, etc, here.